### PR TITLE
Error handling for login

### PR DESCRIPTION
**Pull request by RA341**
## Ui changes

Added specific messages for certain HTTP codes. If a code does not match a specific code, it defaults to the standard error.

err

Added error handling according to todo, a simple dialog for now. 

image

Added a check for empty inputs; if any input fields are empty, this will appear.

image

## bug fixes

There was also a bug in the auth logic 

The if condition adds a port regardless of the protocol, which will cause a tls mismatch error if we put https address and the error handler adds the port.

relevant line

```dart
if (user.serverAdress!.split(":").last != "8096" &&
          user.serverAdress!.split(":").length == 2 
```

adds a HTTP check 
``` dart
if (user.serverAdress!.split(":").last != "8096" &&
          user.serverAdress!.split(":").length == 2 &&
          user.serverAdress!.split(":").first == 'http')
```

```dart
try {
      user = await _apiService.login(
          user.serverAdress!, user.name!, user.password!);
    } catch (e) {
      if (user.serverAdress!.split(":").last != "8096" &&
          user.serverAdress!.split(":").length == 2 &&
          user.serverAdress!.split(":").first == 'http') {
        user.serverAdress = "${user.serverAdress}:8096";
        user = await _apiService.login(
            user.serverAdress!, user.name!, user.password!);
      } else {
        rethrow;
      }
    }
```

### CODE REVIEW

**Comment by sevenrats** [comment]
Hey, thanks! Right off the bat, multiple text boxes is not the way. All jellyfin clients solve this problem, and there are two primary ways of doing it.
1) generate a set of url candidates and dial them, taking the first valid response. This is how things like the roku app do it, but there is the obvious race condition where if you are local to a jf server and its available on both https and http then which one you connect to can sometimes be undefined.
2) generate a set of url candidates and organize them by how preferable they would be, using a complex interaction of delays and timeouts to select the best one. thats how the typescript and kotlin SDK's do it.

File: lib/services/auth_service.dart [hunk 1]
```diff
@@ -60,7 +61,8 @@ class AuthService {
           user.serverAdress!, user.name!, user.password!);
     } catch (e) {
       if (user.serverAdress!.split(":").last != "8096" &&
-          user.serverAdress!.split(":").length == 2) {
+          user.serverAdress!.split(":").length == 2 &&
+          user.serverAdress!.split(":").first == 'http') {
```
Comment (sevenrats) [inline lib/services/auth_service.dart:65]: what if the server address is an ipv6 address with a port

**Comment by sevenrats** [comment]
https://github.com/jellyfin/jellyfin-roku/blob/1c73c7e37b6101487edbe066ceba16de38d5516d/source/utils/misc.bs#L185

**Comment by RA341** [comment]
I am curious as to why. I feel like multiple text boxes make it less error-prone to type in the address.

> Hey, thanks! Right off the bat, multiple text boxes is not the way. All jellyfin clients solve this problem, and there are two primary ways of doing it.
> 
>     1. generate a set of url candidates and dial them, taking the first valid response. This is how things like the roku app do it, but there is the obvious race condition where if you are local to a jf server and its available on both https and http then which one you connect to can sometimes be undefined.
> 
>     2. generate a set of url candidates and organize them by how preferable they would be, using a complex interaction of delays and timeouts to select the best one. thats how the typescript and kotlin SDK's do it.

File: lib/services/auth_service.dart (hunk 1 above)
Comment (RA341) [reply lib/services/auth_service.dart:65]: shouldn't make a difference right, http will always be the first element in the split

**Comment by sevenrats** [comment]
im sorry ive just had this exact conversation too many times. there are at least 3 example of it in the jellyfin git.
in short: its inelegant (not attractive, more touches), its technically uncessary (uri is already a composable and decomposable format), and it presents additional accessibility burdens (people that have difficulty touching will be very frustrated with us because it will stand in the way of the tools they use to interact with software).
Not to mention the elephant in the room: nobody does it.

**Comment by RA341** [comment]
fair enough ill remove it

**Comment by jdk-21** [comment]
I like the error message if there are empty inputs. Can you enhance the the dialog with information for the user about what field is empty.

**Comment by RA341** [comment]
added 

Screenshot_20240828_045012

> I like the error message if there are empty inputs. Can you enhance the the dialog with information for the user about what field is empty.

**Comment by RA341** [comment]
I'll redo the PR

**Pull request closed without merging**